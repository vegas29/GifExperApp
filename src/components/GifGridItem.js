import PropTypes from 'prop-types';

const GifGridItem = ({id, title, url}) => {
    return (
        <div className="bg-white drop-shadow-xl flex flex-col justify-between card animate__animated animate__fadeIn animate__delay-2s">
            <img className="object-cover" src={url} alt={title}/>
            <h3 className="text-center capitalize p-5 text-md">{title}</h3>
            <p className="text-center bg-blue-500 text-white text-lg hover:bg-blue-600"><a href={url} target="_blank" rel="noreferrer">Ver Gif</a></p>
        </div>
    );
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem;