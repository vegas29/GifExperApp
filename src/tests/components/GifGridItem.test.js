import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifExpert',()=>{

    const title = 'Title X';
    const url = 'http://localhost:3000/imagen.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    test('Debe de mostrar el render del componente GifGridItem', ()=>{

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de tener un pÃ¡rrafo con el title', ()=>{

        const h3 = wrapper.find('h3');
        expect(h3.text().trim()).toBe(title);
    })

    test('Debe de tener la imagen igual al url y alt de los props', ()=>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe de tener animate__fadeIn', ()=>{
        const div = wrapper.find('div');
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true);

    })
})