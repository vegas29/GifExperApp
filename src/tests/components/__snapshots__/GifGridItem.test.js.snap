// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="bg-white drop-shadow-xl flex flex-col justify-between card animate__animated animate__fadeIn animate__delay-2s"
  >
    <img
      alt="Eren"
      class="object-cover"
      src="https://shingeki-no-kyogin/eren.jpg"
    />
    <h3
      class="text-center capitalize p-5 text-md"
    >
      Eren
    </h3>
    <p
      class="text-center bg-blue-500 text-white text-lg hover:bg-blue-600"
    >
      <a
        href="https://shingeki-no-kyogin/eren.jpg"
        rel="noreferrer"
        target="_blank"
      >
        Ver Gif
      </a>
    </p>
  </div>
</div>
`;
